doc/guides/coding_guidelines/index.rst

```diff
@@ -823,3 +823,96 @@ Excluding declarations from the header based on compile-time options may prevent
 their documentation from being generated. Their absence also prevents use of
 ``if (IS_ENABLED(CONFIG_FOO)) {}`` as an alternative to preprocessor
 conditionals when the code path should change based on the selected options.
+
+Rule A.2: Inclusive Language
+============================
+
+Severity
+--------
+
+Required
+
+Description
+-----------
+
+Do not introduce new usage of offensive terms listed below. This rule applies
+but is not limited to source code, comments, documentation, and branch names.
+Replacement terms may vary by area or subsystem, but should aim to follow
+updated industry standards when possible.
+
+Exceptions are allowed for maintaining existing implementations of industry
+standard specifications governed externally to the Zephyr Project.
+
+Existing usage is recommended to change as soon as updated industry standard
+specifications become available, or immediately if no specifications apply.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Offensive Terms
+     - Recommended Replacements
+
+   * - ``master / slave``
+     - - ``{primary, main} / {secondary,replica,subordinate}``
+       - ``{initiator,requester} / {target,responder}``
+       - ``{controller,host} / {device,worker,proxy,target}``
+       - ``leader / follower``
+       - ``director / performer``
+       - ``central / peripheral``
+
+   * - ``blacklist / whitelist``
+     - * ``denylist / allowlist``
+       * ``blocklist / passlist``
+
+   * - ``grandfather policy``
+     - TBD
+
+   * - ``sanity``
+     - TBD
+
+Rationale
+---------
+
+Offensive terms do not create an inclusive community environment and therefore
+violate the Zephyr Project `Code of Conduct`_. This coding rule was inspired by
+a similar rule in `Linux`_.
+
+.. _Code of Conduct: https://github.com/zephyrproject-rtos/zephyr/blob/master/CODE_OF_CONDUCT.md
+.. _Linux: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=49decddd39e5f6132ccd7d9fdc3d7c470b0061bb
+
+Status
+------
+
+Related GitHub Issues and Pull Requests are tagged with the `Inclusive Language Label`_.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Area
+     - Selected Replacements
+     - Status
+
+   * - :ref:`bluetooth_api`
+     -
+     -
+
+   * - :ref:`i2c_api`
+     - * ``master / slave`` => ``controller / target``
+     - NXP publishes the `I2C Specification`_ and has selected these
```

Comment:
Has there been a public announcement that these will be the new terms?  (Other than this PR....)

Reply:
No, not yet

Reply:
If we've agreed that we won't be using the names until at least they're officially announced, then I don't think we should document them, because it begs the question of why we aren't updating.

Reply:
If we don't have an announce we can reference to justify using these terms, I think this should be TBD.  Otherwise, we can leave this as is, but add a comment saying that Zephyr will wait for a public announcement before starting to use the replacement terms.

Reply:
I mentioned in the line below that an announcement hasn't been made. Is that ok or do you want a stronger statement?